models/dataset.go

```diff
@@ -0,0 +1,27 @@
+package models
+
+const (
+	fmtEditions                   = "/datasets/%s/editions/%s"
+	fmtEditionsVersions           = "/datasets/%s/editions/%s/versions/%s"
+	fmtEditionsVersionsDimensions = "/datasets/%s/editions/%s/versions/%s/dimensions/%s"
+)
+
+// Datasets represents the model returned from the api datasets
+// endpoint
+type Datasets struct {
+	Items []Dataset
+}
+
+// Dataset represents an individual model dataset
+type Dataset struct {
+	ID             string
+	DimensionLabel string
+	Editions       []DatasetEdition
+}
+
+// DatasetEdition represents an object containing dataset edition links
```

Comment:
This one is worth documenting more clearly, because this is specifically used in the context of a code list, where we're asking the question 'what datasets use this code/list?'

actually, I'd be surprised if that's returning enough info to be meaningful, it should probably have the dataset ID in there too? since we're not formatting the link and returning it, they'll need the dataset ID to build it, because in the code list context the dataset ID isn't implied by the request path in the same way it is in the dataset API.

Reply:
They should have the dataset ID from the `Datasets` or `Dataset` object?

Reply:
Dataset ID is returned higher up in the data structure, see GetCodeDatasets(...) in codelist.go; I have updated comments above the structs